client/settings/payment-settings/account-keys-connection-status.js

```diff
@@ -0,0 +1,180 @@
+`;
+
+export const AccountKeysConnectionStatus = ( { formRef } ) => {
+	const {
+		isTesting,
+		isValid,
+		updateIsTestingAccountKeys,
+		updateIsValidAccountKeys,
+	} = useAccountKeys();
+
+	const handleTestConnection = async ( ref ) => {
+		updateIsTestingAccountKeys( true );
+		updateIsValidAccountKeys( null );
+
+		// Grab the HTMLCollection of elements of the HTML form, convert to array.
+		const elements = Array.from( ref.current.elements );
+		// Convert HTML elements array to an object acceptable for saving keys.
+		const keysToSave = elements.reduce( ( dict, ele ) => {
+			return Object.assign( dict, { [ ele.name ]: ele.value } );
+		}, {} );
+
+		let publishableKey;
+		let secretKey;
+
+		const isTestingLiveConnection =
+			keysToSave.publishable_key && keysToSave.secret_key;
+		const isTestingTestConnection =
+			keysToSave.test_publishable_key && keysToSave.test_secret_key;
+		if ( isTestingLiveConnection ) {
+			publishableKey = keysToSave.publishable_key;
+			secretKey = keysToSave.secret_key;
+		} else if ( isTestingTestConnection ) {
+			publishableKey = keysToSave.test_publishable_key;
+			secretKey = keysToSave.test_secret_key;
+		} else {
+			updateIsTestingAccountKeys( false );
+			updateIsValidAccountKeys( false );
+			return;
+		}
+
+		try {
+			const stripe = await loadStripe( publishableKey );
+			const createTokenResult = await stripe.createToken( 'pii', {
+				personal_id_number: 'connection_test',
+			} );
```

Comment:
Good catch here. I think checking for that error is a good idea.

As for adding a third message, I think the given space is too limited. It may be better if we use a notification instead:

![image](https://user-images.githubusercontent.com/6844516/149174382-babf6554-2df6-498d-8ca7-d67b6a93ffe9.png)

That way, the given space is used specifically to indicate the status of the connection, and the notification is used to expand on details. What do you think?

Reply:
That looks good to me! I'd just uppercase it to HTTPS.